messaging/src/test/java/org/axonframework/queryhandling/PrimitiveQueryHandlerResponseTypeTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.queryhandling;
+
+import org.axonframework.messaging.responsetypes.ResponseTypes;
+import org.axonframework.queryhandling.annotation.AnnotationQueryHandlerAdapter;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests resolving query handlers which return primitive types.
+ */
+public class PrimitiveQueryHandlerResponseTypeTest {
+
+    private final SimpleQueryBus queryBus = SimpleQueryBus.builder().build();
+    private final PrimitiveQueryHandler queryHandler = new PrimitiveQueryHandler();
+    private final AnnotationQueryHandlerAdapter<PrimitiveQueryHandler> annotationQueryHandlerAdapter = new AnnotationQueryHandlerAdapter<>(
+            queryHandler);
+
+    @Before
+    public void setUp() {
+        annotationQueryHandlerAdapter.subscribe(queryBus);
+    }
+
+    @Test
+    public void testInt() {
+        test(0, Integer.class, int.class);
```

Comment:
Did you know you can use parameterized tests to simplify this?
Check out Javadoc on the `org.junit.runners.Parameterized` class.